faasmcli/faasmcli/tasks/sgx.py

```diff
@@ -10,3 +15,35 @@ def check(ctx):
     """
     binary = find_command("detect_sgx")
     run(binary, shell=True)
+
+
+@task
+def build_worker_image(ctx, nocache=False):
+    """
+    Build worker with SGX enabled in hardware mode
+    """
+    # Use buildkit for nicer logging
+    shell_env = copy(environ)
+    shell_env["DOCKER_BUILDKIT"] = "1"
+
+    containers = ["base", "worker"]
+    faasm_ver = get_faasm_version()
+
+    for container in containers:
+        dockerfile = join("docker", "{}.dockerfile".format(container))
+        tag_name = "faasm/{}-sgx:{}".format(container, faasm_ver)
+
+        cmd = [
+            "docker build {}".format("--no-cache" if nocache else ""),
+            "-t {}".format(tag_name),
+            "--build-arg FAASM_VERSION={}".format(faasm_ver),
+            "--build-arg FAASM_SGX_MODE=Hardware",
+            "-f {} .".format(dockerfile),
+        ]
+        docker_cmd = " ".join(cmd)
+        print(docker_cmd)
+        run(docker_cmd, shell=True, check=True, cwd=PROJ_ROOT, env=shell_env)
+
+        docker_push_cmd = "docker push {}".format(tag_name)
+        print(docker_push_cmd)
+        run(docker_push_cmd, shell=True, check=True, cwd=PROJ_ROOT)
```

Comment:
This looks like it's duplicating a lot of logic from the existing container build tasks (https://github.com/faasm/faasm/blob/main/faasmcli/faasmcli/tasks/docker_tasks.py), can we put it in there? It would just need to set the SGX mode conditionally based on the container name, e.g. `sgx_mode = "Hardware" if c.endswith("-sgx") else "Simulation"`, then strip the `-sgx` extension if it was present.